app/controllers/attachments_controller.rb

```diff
@@ -57,19 +57,19 @@ def create
 
   action_auth_level :show, :student
   def show
-    filename = Rails.root.join("attachments", @attachment.filename)
-    unless File.exist?(filename)
-      COURSE_LOGGER.log("Cannot find the file '#{@attachment.filename}' for"\
-                        " attachment #{@attachment.name}")
+    attached_file = @attachment.attachment_file
+    unless attached_file.attached?
+      COURSE_LOGGER.log("No file attached to attachment '#{@attachment.name}'")
 
-      flash[:error] = "Error loading #{@attachment.name} from #{@attachment.filename}"
+      flash[:error] = "No file attached to attachment '#{@attachment.name}'"
       redirect_to([@course, :attachments]) && return
     end
     if @cud.instructor? || @attachment.released?
-      # Set to application/octet-stream to force download
-      send_file(filename, disposition: "inline",
-                          type: "application/octet-stream",
-                          filename: @attachment.filename) && return
+      # Default disposition is "attachment" which forces download
+      send_data attached_file.download,
```

Comment:
bit of an edge case, but if an instructor goes into /storage and deletes a file manually from there, this happens:
![image](https://user-images.githubusercontent.com/50491000/220418025-d35fb406-747c-40ce-9a55-9d1f1032c45c.png)

I don't think this is a huge deal though since it's pretty unlikely & showing the donkey error is reasonable for this.

Reply:
Added error handling

<img width="1026" alt="Screenshot 2023-02-26 at 14 07 29" src="https://user-images.githubusercontent.com/9074856/221431401-4083f05e-6605-4d7e-9f92-5c5a3ff09ade.png">